### ISSUE TITLE
fromJson fails with JPMS

### ISSUE THREAD

**Issue by rendaw** [classgraph/classgraph#311]
```
Exception in thread "main" java.lang.ExceptionInInitializerError
	at my.app/my.app.GUILaunch.<clinit>(GUILaunch.java:63)
	at my.app/my.app.GUIMain.main(GUIMain.java:5)
Caused by: java.lang.IllegalArgumentException: Could not instantiate type class java.net.URI
	at io.github.classgraph@4.6.18/nonapi.io.github.classgraph.json.JSONDeserializer.populateObjectFromJsonObject(JSONDeserializer.java:493)
	at io.github.classgraph@4.6.18/nonapi.io.github.classgraph.json.JSONDeserializer.populateObjectFromJsonObject(JSONDeserializer.java:542)
	at io.github.classgraph@4.6.18/nonapi.io.github.classgraph.json.JSONDeserializer.populateObjectFromJsonObject(JSONDeserializer.java:542)
	at io.github.classgraph@4.6.18/nonapi.io.github.classgraph.json.JSONDeserializer.deserializeObject(JSONDeserializer.java:613)
	at io.github.classgraph@4.6.18/nonapi.io.github.classgraph.json.JSONDeserializer.deserializeObject(JSONDeserializer.java:640)
	at io.github.classgraph@4.6.18/io.github.classgraph.ScanResult.fromJSON(ScanResult.java:1016)
	at my.app/my.app.ConfigBase.<clinit>(ConfigBase.java:29)
	... 2 more
Caused by: java.lang.IllegalAccessException: class nonapi.io.github.classgraph.json.JSONDeserializer (in module io.github.classgraph) cannot access a member of class java.net.URI (in module java.base) with modifiers "private"
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:361)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:591)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at io.github.classgraph@4.6.18/nonapi.io.github.classgraph.json.JSONDeserializer.populateObjectFromJsonObject(JSONDeserializer.java:482)
	... 8 more
```

It looks like I might need to do --add-opens for java.base?  I'm adding this as an issue since it seems strange that deserializing would need it when regular scanning does not (trying to do the build time scan since runtime scanning doesn't seem to be working in a jlinked image).

**Comment by lukehutch** [comment]
Thanks for the report. I saw this bug recently, and it should already be fixed in master. Can you please test the master version?

**Comment by rendaw** [comment]
Wow, this project moves fast.  Sorry, I should have checked for updates first.  This works with 4.6.32.

**Comment by lukehutch** [comment]
@rendaw yeah, I try to be really responsive to bug reports, and I try to make sure people get fixes in no more than a day or two, so that translates into a lot of small releases. Thanks for testing!

**Comment by lukehutch** [comment]
Released in 4.8.0.

https://github.com/classgraph/classgraph/releases/tag/classgraph-4.8.0

Thanks for the bug report!
